Add support for configurable interleave modes

dougyau opened the pull request:
Introduce the `InterleaveMode` enum to define encoding/decoding memory arrangements. Implement a `set_interleave_mode` method in the `CharLS` struct to allow setting the interleave mode for the encoder. This enhances flexibility when handling multi-component images.

dougyau committed:
Add support for configurable interleave modes

Introduce the `InterleaveMode` enum to define encoding/decoding memory arrangements. Implement a `set_interleave_mode` method in the `CharLS` struct to allow setting the interleave mode for the encoder. This enhances flexibility when handling multi-component images.

dougyau committed:
Simplify return logic in JPEG-LS encoder setup

Removed unnecessary `Ok(())` wrapping, directly returning the result of `translate_error(err)`. This improves code readability and aligns with Rust's idiomatic practices.

Enet4 commented on src/lib.rs:
```diff
@@ -76,6 +76,31 @@ pub struct FrameInfo {
+///
+/// Defines how color components are arranged in memory for both native and encoded formats.
+/// See more: https://github.com/team-charls/charls/wiki/How-to-use-the-API#multi-component-image-typical-rgb-color
+#[derive(Debug, Default)]
```
Other useful traits which can be derived here: `Clone`, `Copy`, `PartialEq`, `Eq`, `Hash`

dougyau committed:
Add additional derives to InterleaveMode enum

Implemented Clone, Copy, PartialEq, Eq, and Hash for InterleaveMode to enhance usability and compatibility with standard Rust traits. This change supports more flexible usage scenarios, such as using InterleaveMode in collections or comparisons.

dougyau commented:
Resolves #5

dougyau committed:
Update crate version to 0.4.0 in Cargo files

Bump the crate version from 0.3.1 to 0.4.0 in Cargo.toml and Cargo.lock. This change reflects a new release, likely including new features, fixes, or other updates. Ensure dependencies remain in sync with the updated version.
